h/api/auth.py

```diff
@@ -13,3 +15,21 @@ def get_user(request):
                 consumer = Consumer(key)
                 return User(userid, consumer, False)
     return None
+
+
+def translate_annotation_principals(principals):
+    """
+    Translate a list of annotation principals to a list of pyramid principals.
+    """
+    result = set([])
+    for principal in principals:
+        # Ignore suspicious principals from annotations
+        if principal.startswith('system.'):
+            continue
+        if principal == 'group:__world__':
+            result.add(security.Everyone)
```

Comment:
That's the definition of `group:__world__` ... and yes, `'update': ['group:__world__']` would allow anyone to update the annotation, just as `'read': ['group:__world__']` allows anyone to read the annotation, even if they're not logged in.

Also, as you note, this isn't a change in behaviour. This function has just been factored out.